wdl-format/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+        write!(writer, "{}", self.as_str())
+    }
+}
+
+impl Formattable for VersionStatement {
+    fn format<T: std::fmt::Write>(&self, writer: &mut T, state: &mut State) -> std::fmt::Result {
+        let mut preceding_comments = Vec::new();
+        let comment_buffer = &mut String::new();
+        for sibling in self.syntax().siblings_with_tokens(Direction::Prev) {
+            match sibling.kind() {
+                SyntaxKind::Comment => {
+                    let comment = Comment::cast(
+                        sibling
+                            .as_token()
+                            .expect("Comment should be a token")
+                            .clone(),
+                    )
+                    .expect("Comment should cast to a comment");
+                    comment.format(comment_buffer, state)?;
+                    preceding_comments.push(comment_buffer.clone());
+                    comment_buffer.clear();
+                }
+                SyntaxKind::Whitespace => {
+                    // Ignore
+                }
+                SyntaxKind::VersionStatementNode => {
+                    // Ignore the root node
+                }
+                _ => {
+                    unreachable!("Unexpected syntax kind: {:?}", sibling.kind());
+                }
+            }
+        }
+
+        for comment in preceding_comments.iter().rev() {
+            write!(writer, "{}", comment)?;
+        }
+
+        // If there are preamble comments, ensure a blank line is inserted
+        if !preceding_comments.is_empty() {
+            write!(writer, "{}", NEWLINE)?;
+        }
+
+        let version_keyword = self
+            .syntax()
+            .children_with_tokens()
+            .find(|element| element.kind() == SyntaxKind::VersionKeyword)
+            .expect("Version statement should have a version keyword");
+        write!(writer, "{}", version_keyword)?;
+        format_inline_comment(&version_keyword, writer, state, true)?;
+
+        let version = self.version();
+
+        format_preceding_comments(
+            &SyntaxElement::from(version.syntax().clone()),
+            writer,
+            state,
+            true,
+        )?;
+        state.space_or_indent(writer)?;
+        version.format(writer, state)?;
+        format_inline_comment(
+            &SyntaxElement::from(self.syntax().clone()),
+            writer,
+            state,
+            false,
+        )
+    }
+}
+
+impl Formattable for Ident {
+    fn format<T: std::fmt::Write>(&self, writer: &mut T, _state: &mut State) -> std::fmt::Result {
+        write!(writer, "{}", self.as_str())
+    }
+}
+
+/// Find an expected child element of the specified kind.
+///
+/// # Panics
+/// Panics if the child element is not found.
+pub fn first_child_of_kind(node: &SyntaxNode, kind: SyntaxKind) -> SyntaxElement {
+    node.children_with_tokens()
+        .find(|element| element.kind() == kind)
+        .expect("Expected to find a child")
```

Comment:
Maybe add the `kind` to the `expect` message.

Reply:
I don't think there's a way to do that. When I try `.expect("Expected to find a child of kind: {:?}", kind)` I get a compiler error: `this method takes 1 argument but 2 arguments were supplied`

Reply:
The only way would be to use `unwrap_or_else(|| panic!("Expected to find a child of kind: {kind:?}"))`.

Reply:
Done

Reply:
Thanks @peterhuene!

Reply:
@peterhuene  - Is `unwrap_or_else(|| panic!("Expected to find a child of kind: {kind:?}"))` the more Rust way? I would have done `.expect(&format!("Expected to find a child of kind: {:?}", kind))`.